### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by rashfael** [Shinigami92/eslint-plugin-vue-pug-sfc#157] [General]
Hi!
I've been using pug and vue for a long time and I'd love to contribute to getting pug to lint in vue SFCs.

I have read through existing discussions like https://github.com/vuejs/vue-eslint-parser/issues/29 and played around with some code.

If I may ask, what made you go with a separate plugin and copying rules instead of integrating pug with vue-eslint-parser?

**Comment by Shinigami92** [comment]
> If I may ask, what made you go with a separate plugin and copying rules instead of integrating pug with vue-eslint-parser?

1. The authors didn't want to extend it to support pug additionally to pug, because it would make double of work
2. The kind of how the lexer of pug works differentiates much from how it works of the html stuff

So right now, we need to do everything twice :slightly_frowning_face: one for html and one for pug
There were one try of making something to generate html out of the pug and use the html parser, but that dramatically failed and had many problems
You are loosing stuff like `--fix` for pug and so on. And it's even more complicated than this plugin is right now

**Reply by Shinigami92** [reply]
@rashfael But contributing from you would be really be helpful and I can try to help as much as possible to help your in onboarding.
I really need the support of contributors in this repository. I even think about to give some money via GH-sponsor or PayPal for successfully implemented rules.

**Reply by rashfael** [reply]
> 1. The authors didn't want to extend it to support pug additionally to pug, because it would make double of work

Ah, I feared as much. That of course basically rules out solving pug linting in vue-eslint-parser.

> There were one try of making something to generate html out of the pug and use the html parser, but that dramatically failed and had many problems You are loosing stuff like `--fix` for pug and so on. And it's even more complicated than this plugin is right now.

Yeah, I had a look at that today and got most of the original `attribute-hyphenation` rule to run with modifying vue-eslint-parser, but simulating html tokens has a lot of problems, the biggest conceptual one is probably not actually be able to lint pug-specific syntax (like commas after attributes and things like that).

I'd be happy to contribute to this project. Should I just pick an issue with Priority A to start off?
I'm especially interested in getting https://github.com/Shinigami92/eslint-plugin-vue-pug-sfc/issues/8 to work, since that interferes with linting `<setup script>` quite a bit, but I see you already started a PR for that :)

**Reply by Shinigami92** [reply]
PLEASE fork the repo and make a PR based on https://github.com/Shinigami92/eslint-plugin-vue-pug-sfc/tree/rule/script-setup-uses-vars :heart: 
I'm not really actively working on that right now so you can start from my base I setup there and therefore can start with actually implementing the rule

I already got headaches on thinking of implementing this rule, cause it works together with the script block + we do not have access to functions that are already defined internally in `eslint-plugin-vue` :slightly_frowning_face:, so we need to copy and write our own stuff

**Reply by rashfael** [reply]
Thanks, I'll have a look at it!

**Reply by Shinigami92** [reply]
I will pay you 10€ if you successfully implement this rule and I can release it. That's how much the rule is worth right now.
To put in relation: I think about to pay 1-3 € per rule that I think it is worth to pay.
Some rules are too easy to implement to get payed for, but some are more complicated, because e.g. they need to access sub-scoped variables in pug code, and I currently do not have a parser or parsed out code for that embedded JS stuff right now.

Do not see it as a money per hour, because the "minimum wage" would be worth as hell :wink:

**Reply by rashfael** [reply]
Thanks for the offer, but for me having a usable linter for my favorite preprocessor is payment enough.
Since this will be useful for my whole team at my employer I will probably develop this on company time anyways :D

**Reply by Shinigami92** [reply]
I could also invite you as collaborator into the repo if you are interested, you could be a co-maintainer in the long run if your PR is successful

**Reply by Shinigami92** [reply]
@rashfael Oh and as a side-note: the labels do not reflect the priority but the "kind" of what the rule is defined by `eslint-plugin-vue`
<img src="https://user-images.githubusercontent.com/7195563/158037334-53a6845e-080d-4183-ad86-1f0f51d79089.png" width="200px"/>

So go with your own defined prio, the rules that are requested by us - the community - are the ones that should be prioritized first, because that are the rules where we benefit the most from.

**Reply by rashfael** [reply]
Ah, I see, thanks for clarifying.
